shell/platform/windows/platform_handler.cc

```diff
@@ -340,11 +354,61 @@ void PlatformHandler::SystemSoundPlay(
   }
 }
 
+void PlatformHandler::SystemExitApplication(
+    const std::string& exit_type,
+    int64_t exit_code,
+    std::unique_ptr<MethodResult<rapidjson::Document>> result) {
+  rapidjson::Document result_doc;
+  result_doc.SetObject();
+  if (exit_type.compare(kExitTypeRequired) == 0) {
+    QuitApplication(exit_code);
+    result_doc.GetObjectW().AddMember(kExitResponseKey, kExitResponseExit,
+                                      result_doc.GetAllocator());
+    result->Success(result_doc);
+  } else {
+    exit_code_ = exit_code;
```

Comment:
Instead of storing this, you could maybe pass it to `RequestAppExit`, and then capture it in the callback instead of storing it here.  Then you won't need to initialize it in the constructor, or clear it when you get a response. The exit code could be an optional argument to `RequestAppExit`, defaulting to zero. You'd probably have to pass it to `RequestAppExitSuccess` too.